lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -709,18 +710,67 @@ void convert_ambr(
   *ambr_value = static_cast<uint16_t>(temp_pdu_ambr_response_value);
 }
 
+// Utility function to convert address to buffer
+static int paa_to_address_info(
+    const paa_t* paa, uint8_t* pdu_address_info, uint8_t* pdu_address_length) {
+  uint32_t ip_int = 0;
+  if ((paa == nullptr) || (pdu_address_info == nullptr) ||
+      (pdu_address_length == nullptr)) {
+    return RETURNerror;
+  }
+  switch (paa->pdn_type) {
+    case IPv4:
+      ip_int = ntohl(paa->ipv4_address.s_addr);
+      INT32_TO_BUFFER(ip_int, pdu_address_info);
+      *pdu_address_length = sizeof(ip_int);
+      break;
+
+    case IPv6:
+      if (paa->ipv6_prefix_length == IPV6_PREFIX_LEN) {
+        memcpy(
+            pdu_address_info, &paa->ipv6_address.s6_addr[IPV6_INTERFACE_ID_LEN],
+            paa->ipv6_prefix_length / 8);
+        *pdu_address_length = IPV6_INTERFACE_ID_LEN;
```

Comment:
wouldn't it be 
memcpy(
            pdu_address_info, &paa->ipv6_address.s6_addr[IPV6_PREFIX_LEN/8],
            IPV6_INTERFACE_ID_LEN);
?

Reply:
Just a comment that we can skip here but what seems confusing is that PREFIX unit is in bits and interface ID in bytes , would be good to harmonize or add a suffix ?

Reply:
Addressed the review comment